File: wrappers/mvc/demos/Kendo.Mvc.Examples/Models/CategoryItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Kendo.Mvc.Examples.Models
{
    public class CategoryItem
    {
        public string CategoryName { get; set; }
        public List<SubCategoryItem> SubCategories { get; set; }
    }
}

File: wrappers/java/spring-demos/src/main/java/com/kendoui/spring/models/CategoryItem.java
package com.kendoui.spring.models;

import java.util.ArrayList;

public class CategoryItem {
    private String categoryName;
    private ArrayList<SubCategoryItem> subCategories;       
    
    public String getCategoryName()
    {
        return this.categoryName;
    }    
    
    public void setCategoryName(String value)
    {
        this.categoryName = value;
    }
    
    public ArrayList<SubCategoryItem> getSubCategories()
    {        
        return this.subCategories;
    }
    
    public SubCategoryItem AddSubItem()
    {               
        SubCategoryItem item = new SubCategoryItem();
        if(this.subCategories == null){
            this.subCategories = new ArrayList<SubCategoryItem>();
        }
        
        this.subCategories.add(item);
        return item;
    }  
}
